File: javascript/bob/bob.js
(function() {
  "use strict";

  function Bob() {
    function isSilent(phrase) {
      return phrase === '';
    }

    function isShouting(phrase) {
      return phrase.toUpperCase() === phrase;
    }

    function isAskingAQuestion(phrase) {
      return phrase[phrase.length - 1] === '?';
    }

    this.hey = function(phrase) {
      switch (true) {
        case isSilent(phrase):
          return 'Fine, be that way.';
        case isShouting(phrase):
          return 'Woah, chill out!';
        case isAskingAQuestion(phrase):
          return 'Sure';
        default:
          return 'Whatever';
      }
    };
  }

  module.exports = Bob;
})();

File: clojure/bob/bob.clj
(ns bob
  (require [clojure.string :as string]))

(defn- question? [phrase] (= \? (last phrase)))
(defn- shout?    [phrase] (= phrase (string/upper-case phrase)))
(defn- silence?  [phrase] (string/blank? phrase))

(defn response-for [phrase]
  (cond
    (silence?  phrase) "Fine. Be that way!"
    (shout?    phrase) "Woah, chill out!"
    (question? phrase) "Sure."
    :else              "Whatever."))